src/data-dictionary/IgSummary.ts

```diff
@@ -263,9 +263,25 @@ export class IgSummary {
             logger.error(`No profiles found in ${this.fhirDefFolder}.`);
             return;
         }
+
+        let usedByMeasureMap = new Map<string, string>();
+        let assoWithVSMap = new Map<string, string>();
         for (const sd of _.uniqBy(this.defs.allProfiles(), 'id')) {
             const snapshot = sd.snapshot;
-
+            const profileTitle = sd.title || sd.name;
+            // usedByMeasureMap and assoWithVSMap are temporary. Eventually extension flag can be included in DataElementInformation
+            for (const elemJson of snapshot.element.slice(1)) {
+                let extension_flg:boolean = false;
+                elemJson.extension?.forEach((ext: any) => {
+                    if (ext.url.includes('/StructureDefinition/used-by-measure')) extension_flg = true;
```

Comment:
I think we can easily generalize this to allow the tool to look for arbitrary extensions:
1. add a configuration setting for the extension url (or snippet) and use that in place of the literal string `'/StructureDefinition/used-by-measure'` in this line
2. add a configuration setting for the column name.

With that change, plus some documentation, I would feel comfortable merging this into the main ig-summary branch.

Reply:
https://github.com/mitre/ig-summary/pull/3/commits/db51b61206bbc2cdb64c5b4f7360a8f21364fe56

In this commit, `extensionColumn` is added in the configuration. IG summary create a column `Extension?` as long as there's configuration. Though it is ideal to have the column header dynamic (such as `Used by measure?`) it is based on `enum` and can't change, so I just put `Extension?` and captured extension name in the main summary tab.

Reply:
If I understand correctly, you're saying that the column name will always be "Extension?" with a user-supplied description on another tab?

If so, we can and should do better than that. Potential places to look (based on a very brief look at the code):
1. Enums are just strings under the covers - do you really need to have the column name in an enum, or after its gone through the enum, can you add something dynamic after it?
2. We control spreadsheet generation, when generating column headers, can you identify the "Extension?" one and replace it with a string from the configuration?

Reply:
My initial approach is to make it dynamic, including customized column names and multiple extension flags, etc. The problem is it is very limited while column setting is tied with `ExcelJS.Worksheet` and it that it has to be enum. And to my best knowledge enum can't be dynamic in run time. So I need to figure out if there's any flexibility to change the Excel library IG summary is using. I'm not saying it's not possible, but it will be extra work than I thought.

Reply:
At any rate, I will try to see how touching `ExcelJS.Worksheet` would add complexity.

Reply:
https://github.com/mitre/ig-summary/pull/3/commits/e65e03d5290aaf506b804da6909127ee0397669e

Made changes to enable dynamic extension flag columns. Test with `--settings data_dictionary_settings.example.yaml` option.

Reply:
looks good